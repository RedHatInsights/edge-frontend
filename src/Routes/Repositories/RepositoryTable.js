import React, { useState } from 'react';
import ToolbarHeader from './ToolbarHeader';
import ToolbarFooter from './ToolbarFooter';
import Table from './Table';
import PropTypes from 'prop-types';

const RepositoryTable = ({ data, toolbarButtons, toggle }) => {
  const [input, setInput] = useState('');
  const [perPage, setPerPage] = useState(10);
  const [page, setPage] = useState(1);
  const filteredByName = () =>
    data
      .filter((repo) => repo.name.toLowerCase().includes(input.toLowerCase()))
      .reverse();
  return (
    <>
      <ToolbarHeader
        count={data.length}
        toolbarButtons={toolbarButtons}
        setInput={setInput}
        perPage={perPage}
        setPerPage={setPerPage}
        page={page}
        setPage={setPage}
      />
      <Table
        toggle={toggle}
        columns={['Name']}
        rows={filteredByName().slice(
          (page - 1) * perPage,
          (page - 1) * perPage + perPage
        )}
      />
      <ToolbarFooter
        count={data.length}
        setInput={setInput}
        perPage={perPage}
        setPerPage={setPerPage}
        page={page}
        setPage={setPage}
      />
    </>
  );
};
RepositoryTable.propTypes = {
  data: PropTypes.array,
  toolbarButtons: PropTypes.array,
  toggle: PropTypes.func,
};

export default RepositoryTable;
