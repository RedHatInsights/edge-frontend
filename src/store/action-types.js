import keyBy from 'lodash/keyBy';
import flatMap from 'lodash/flatMap';

export const LOAD_DEVICE_SUMMARY = 'LOAD_DEVICE_SUMMARY';
export const LOAD_ACTIVE_IMAGES = 'LOAD_ACTIVE_IMAGES';
export const LOAD_EDGE_IMAGES = 'LOAD_EDGE_IMAGES';
export const LOAD_EDGE_IMAGE_SETS = 'LOAD_EDGE_IMAGE_SETS';
export const LOAD_EDGE_IMAGE_PACKAGES = 'LOAD_EDGE_IMAGE_PACKAGES';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_GROUP_DETAIL = 'LOAD_GROUP_DETAIL';
export const LOAD_TRESHOLD = 'LOAD_TRESHOLD';
export const LOAD_DEVICES_INFO = 'LOAD_DEVICES_INFO';
export const LOAD_CANARIES_INFO = 'LOAD_CANARIES_INFO';
export const LOAD_GROUP_DEVICES_INFO = 'LOAD_GROUP_DEVICES_INFO';
export const LOAD_IMAGE_STATUS = 'LOAD_IMAGE_STATUS';
export const LOAD_IMAGE_DETAIL = 'LOAD_IMAGE_DETAIL';
export const LOAD_IMAGE_SET_DETAIL = 'LOAD_IMAGE_SET_DETAIL';
export const CREATE_NEW_IMAGE = 'CREATE_NEW_IMAGE';
export const CREATE_NEW_IMAGE_RESET = `${CREATE_NEW_IMAGE}_RESET`;
export const POLLING_IMAGES = 'POLLING_IMAGES';
export const LOAD_DEVICE_TABLE = 'LOAD_DEVICE_TABLE';
export const UPDATE_DISPLAY_NAME = 'UPDATE_DISPLAY_NAME';
export const REMOVE_ENTITY = 'REMOVE_ENTITY';

const asyncActions = flatMap(
  [
    LOAD_EDGE_IMAGES,
    LOAD_EDGE_IMAGE_SETS,
    LOAD_EDGE_IMAGE_PACKAGES,
    LOAD_ACTIVE_IMAGES,
    LOAD_GROUPS,
    LOAD_GROUP_DETAIL,
    LOAD_TRESHOLD,
    LOAD_DEVICES_INFO,
    LOAD_CANARIES_INFO,
    LOAD_GROUP_DEVICES_INFO,
    LOAD_IMAGE_STATUS,
    LOAD_IMAGE_DETAIL,
    LOAD_IMAGE_SET_DETAIL,
    CREATE_NEW_IMAGE,
    LOAD_DEVICE_TABLE,
    UPDATE_DISPLAY_NAME,
    REMOVE_ENTITY,
  ],
  (a) => [a, `${a}_PENDING`, `${a}_FULFILLED`, `${a}_REJECTED`]
);
export const ACTION_TYPES = keyBy(asyncActions, (k) => k);

export const SELECT_ENTITY = 'SELECT_ENTITY';
export const PRE_SELECT_ENTITY = 'PRE_SELECT_ENTITY';
export const CLEAN_ENTITIES = 'CLEAN_ENTITIES';
