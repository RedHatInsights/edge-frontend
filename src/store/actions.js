import {
  ACTION_TYPES,
  LOAD_TRESHOLD,
  LOAD_DEVICES_INFO,
  LOAD_CANARIES_INFO,
  LOAD_GROUP_DETAIL,
  LOAD_GROUP_DEVICES_INFO,
  SELECT_ENTITY,
  PRE_SELECT_ENTITY,
  CLEAN_ENTITIES,
  LOAD_ACTIVE_IMAGES,
  LOAD_EDGE_IMAGES,
  LOAD_EDGE_IMAGE_SETS,
  LOAD_EDGE_IMAGE_PACKAGES,
  LOAD_DEVICE_SUMMARY,
  LOAD_IMAGE_STATUS,
  LOAD_IMAGE_DETAIL,
  LOAD_IMAGE_SET_DETAIL,
  CREATE_NEW_IMAGE,
  POLLING_IMAGES,
  LOAD_DEVICE_TABLE,
} from './action-types';

import {
  fetchImageStatus,
  getImageById,
  fetchEdgeImages,
  fetchEdgeImageSets,
  getImagePackageMetadata,
  createImage,
  getImageSet,
  fetchActiveImages,
} from '../api/images';
import { getInventory } from '../api/devices';
import { hosts } from '../api';
import { deleteSystemsById } from '../utils';

export const loadThreshold = () => ({
  type: LOAD_TRESHOLD,
  payload: () => {},
});

export const loadDevicesInfo = () => ({
  type: LOAD_DEVICES_INFO,
  payload: () => {},
});

export const loadCanariesInfo = () => ({
  type: LOAD_CANARIES_INFO,
  payload: () => {},
});

export const loadGroupsDetail = () => ({
  type: LOAD_GROUP_DETAIL,
  payload: () => {},
});

export const loadGroupDevicesInfo = () => ({
  type: LOAD_GROUP_DEVICES_INFO,
  payload: () => {},
});

export const selectEntity = (id, selected) => ({
  type: SELECT_ENTITY,
  payload: {
    id,
    selected,
  },
});

export const preSelectEntity = (id, selected) => ({
  type: PRE_SELECT_ENTITY,
  payload: {
    id,
    selected,
  },
});

export const cleanEntities = () => ({
  type: CLEAN_ENTITIES,
});

export const loadImages = (dispatch, pagination) => {
  dispatch({
    type: LOAD_ACTIVE_IMAGES,
    payload: fetchActiveImages(pagination),
    meta: {
      notifications: {
        rejected: {
          variant: 'danger',
          title: 'Cannot show images data',
          description: 'Failed receiving images from image-builder',
        },
      },
    },
    // the '.catch' part is necessary because redux-promise-middleware throws the error on REJECTED
    // and to avoid the app exploding I need to catch it here.
    // THANK you redux-promise-middleware for not allowing to customize this behavior. ðŸ˜ 
  }).catch(() => null);
};

export const loadDeviceSummary = (dispatch) => {
  dispatch({
    type: LOAD_DEVICE_SUMMARY,
    payload: () => {},
    meta: {
      notifications: {
        rejected: {
          variant: 'danger',
          title: 'Cannot show system summary data',
          description: 'Failed receiving system summary data from inventory',
        },
      },
    },
    // the '.catch' part is necessary because redux-promise-middleware throws the error on REJECTED
    // and to avoid the app exploding I need to catch it here.
    // THANK you redux-promise-middleware for not allowing to customize this behavior. ðŸ˜ 
  }).catch(() => null);
};

export const loadImageStatus = (dispatch, imageId) => {
  dispatch({
    type: LOAD_IMAGE_STATUS,
    payload: fetchImageStatus({ id: imageId }),
  }).catch(() => null);
};

export const loadImageDetail = (dispatch, imageId) => {
  dispatch({
    type: LOAD_IMAGE_DETAIL,
    payload: getImageById({ id: imageId }),
  }).catch(() => null);
};

export const createNewImage = (dispatch, payload, callback) => {
  dispatch({
    type: CREATE_NEW_IMAGE,
    payload: createImage(payload),
  })
    .then(callback)
    .catch(() => null);
};

export const loadEdgeImages = (dispatch, query) => {
  dispatch({
    type: LOAD_EDGE_IMAGES,
    payload: fetchEdgeImages(query),
  }).catch(() => null);
};

export const loadEdgeImageSets = (dispatch, query) => {
  dispatch({
    type: LOAD_EDGE_IMAGE_SETS,
    payload: fetchEdgeImageSets(query),
  }).catch(() => null);
};

export const setPolling = (toStart, interval) => {
  const subAction = toStart ? 'START' : 'END';
  const payload = toStart ? { interval } : {};
  return {
    type: `${POLLING_IMAGES}_${subAction}`,
    ...payload,
  };
};

export const addImageToPoll = ({ id, name }) => {
  return {
    type: `${POLLING_IMAGES}_ADD`,
    payload: {
      name,
      id,
    },
  };
};

export const removeImagesToPoll = (ids) => {
  return {
    type: `${POLLING_IMAGES}_REMOVE`,
    ids,
  };
};

export const loadImageSetDetail = (dispatch, urlParam, query) => {
  dispatch({
    type: LOAD_IMAGE_SET_DETAIL,
    payload: getImageSet({ id: urlParam, q: query }),
  }).catch(() => null);
};

export const loadImagePackageMetadata = (dispatch, imageId) => {
  dispatch({
    type: LOAD_EDGE_IMAGE_PACKAGES,
    payload: getImagePackageMetadata(imageId),
  }).catch(() => null);
};

export const loadDeviceTable = (dispatch) => {
  dispatch({
    type: LOAD_DEVICE_TABLE,
    payload: getInventory(),
  }).catch(() => null);
};

export const editDisplayName = (id, value, origValue) => ({
  type: ACTION_TYPES.UPDATE_DISPLAY_NAME,
  payload: hosts.apiHostPatchHostById([id], { display_name: value }),
  meta: {
    id,
    value,
    origValue,
    notifications: {
      fulfilled: {
        variant: 'success',
        title: `Display name for entity with ID ${id} has been changed to ${value}`,
        dismissable: true,
      },
    },
  },
});

export const deleteEntity = (systems, displayName) => ({
  type: ACTION_TYPES.REMOVE_ENTITY,
  payload: deleteSystemsById(systems),
  meta: {
    notifications: {
      fulfilled: {
        variant: 'success',
        title: 'Delete operation finished',
        description: `${displayName} has been successfully removed.`,
        dismissable: true,
      },
    },
    systems,
  },
});
