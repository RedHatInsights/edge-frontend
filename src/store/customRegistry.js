import ReducerRegistry from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';

export const createRegistry = (
  initialState = {},
  middleware = [],
  composeEnhancersDefault
) => {
  return new ReducerRegistry(
    initialState,
    [...middleware],
    composeEnhancersDefault
  );
};
