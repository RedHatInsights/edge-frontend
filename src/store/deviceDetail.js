import GeneralInformationTab from '../components/DeviceDetail';
import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';

const entityLoaded = (state) => {
  return {
    ...state,
    loaded: true,
    activeApps: [
      {
        title: 'General information',
        name: 'general_information',
        component: GeneralInformationTab,
      },
    ],
  };
};

export const deviceDetail = applyReducerHash({
  LOAD_ENTITY_FULFILLED: entityLoaded,
});
