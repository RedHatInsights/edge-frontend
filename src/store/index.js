import { createContext } from 'react';
import { ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import { polling } from './polling';

export const RegistryContext = createContext({
  getRegistry: () => {},
});

export function init(...middleware) {
  return new ReducerRegistry({}, [
    promiseMiddleware,
    notificationsMiddleware({
      errorDescriptionKey: ['detail', 'stack'],
    }),
    polling(),
    ...middleware,
  ]);
}
