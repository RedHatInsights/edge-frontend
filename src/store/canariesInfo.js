import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { LOAD_CANARIES_INFO } from './action-types';

const initialState = {};

const loadCanariesInfoPending = (state) => ({
  ...state,
  canariesInfo: [],
  isLoading: true,
});
const loadCanariesInfoFulfilled = (state, { payload }) => ({
  ...state,
  canariesInfo: payload?.results || [],
  isLoading: false,
});

export default applyReducerHash(
  {
    [`${LOAD_CANARIES_INFO}_PENDING`]: loadCanariesInfoPending,
    [`${LOAD_CANARIES_INFO}_FULFILLED`]: loadCanariesInfoFulfilled,
  },
  initialState
);
