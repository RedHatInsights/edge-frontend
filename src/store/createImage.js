import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { CREATE_NEW_IMAGE } from './action-types';

const initialState = {};

const createNewImagePending = () => ({
  isLoading: true,
  hasError: false,
  error: null,
});

const createNewImageFulfilled = () => {
  return {
    isLoading: false,
    hasError: false,
    error: null,
  };
};

const createNewImageRejected = (_state, { payload }) => ({
  isLoading: false,
  hasError: true,
  error: payload,
});

const createNewImageReset = () => ({
  isLoading: false,
  hasError: false,
  error: null,
});

export default applyReducerHash(
  {
    [`${CREATE_NEW_IMAGE}_PENDING`]: createNewImagePending,
    [`${CREATE_NEW_IMAGE}_FULFILLED`]: createNewImageFulfilled,
    [`${CREATE_NEW_IMAGE}_REJECTED`]: createNewImageRejected,
    [`${CREATE_NEW_IMAGE}_RESET`]: createNewImageReset,
  },
  initialState
);
