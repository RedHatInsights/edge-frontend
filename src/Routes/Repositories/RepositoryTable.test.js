import React from 'react';
import { render, screen } from '@testing-library/react';
import RepositoryTable from './RepositoryTable';
import { init, RegistryContext } from '../../store';
import logger from 'redux-logger';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

describe('RepositoryTable', () => {
  it('renders correctly', () => {
    const registry = init(logger);

    const data = [
      {
        ID: 1,
        Name: 'test name 1',
        URL: 'test url 1',
      },
      {
        ID: 2,
        Name: 'test name 2',
        URL: 'test url 2',
      },
      {
        ID: 3,
        Name: 'test name 3',
        URL: 'test url 3',
      },
    ];

    const closeModal = jest.fn();

    const { container } = render(
      <RegistryContext.Provider
        value={{
          getRegistry: () => registry,
        }}
      >
        <Provider store={registry.getStore()}>
          <RepositoryTable
            isLoading={false}
            hasError={false}
            data={data}
            count={data.length}
            closeModal={closeModal}
            fetchRepos={() => data}
          />
        </Provider>
      </RegistryContext.Provider>,
      { wrapper: MemoryRouter }
    );

    expect(
      screen.getByLabelText('Select input for name').getAttribute('placeholder')
    ).toEqual('Filter by name');
    expect(screen.findByRole('button', 'Add repository')).toBeDefined();
    expect(screen.findByRole('button', 'Name')).toBeDefined();
    expect(screen.getAllByText(/test name/i)).toHaveLength(3);
    expect(screen.getAllByRole('link', /test url/i)).toHaveLength(3);

    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
