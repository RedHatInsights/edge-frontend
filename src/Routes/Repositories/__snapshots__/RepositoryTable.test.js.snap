// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTable renders correctly 1`] = `
<div
  class="pf-v5-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-1"
  data-ouia-component-type="PF5/Toolbar"
  data-ouia-safe="true"
  data-testid="toolbar-header-testid"
  id="toolbar-header"
  style="padding: 0px;"
>
  <div
    class="pf-v5-c-toolbar__content"
  >
    <div
      class="pf-v5-c-toolbar__content-section"
    >
      <div
        class="pf-v5-c-toolbar__item"
        data-testid="filter-input-testid"
      >
        <div
          class="pf-v5-c-input-group"
        >
          <div
            class="pf-v5-c-input-group__item"
          >
            <div
              class="pf-v5-c-text-input-group"
              id="textInput1"
              type="search"
            >
              <div
                class="pf-v5-c-text-input-group__main pf-m-icon"
              >
                <span
                  class="pf-v5-c-text-input-group__text"
                >
                  <span
                    class="pf-v5-c-text-input-group__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      />
                    </svg>
                  </span>
                  <input
                    aria-label="Select input for name"
                    class="pf-v5-c-text-input-group__text-input"
                    name="textInput1"
                    placeholder="Filter by name"
                    type="text"
                    value=""
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pf-v5-c-toolbar__item"
      >
        <button
          aria-disabled="false"
          class="pf-v5-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          type="button"
        >
          Add repository
        </button>
      </div>
      <div
        class="pf-v5-c-toolbar__item pf-m-pagination pf-m-align-right"
      >
        <div
          class="pf-v5-c-pagination pf-m-compact"
          data-ouia-component-id="OUIA-Generated-Pagination-top-1"
          data-ouia-component-type="PF5/Pagination"
          data-ouia-safe="true"
          data-testid="pagination-header-test-id"
          id="pagination-options-menu-top-top-pagination"
          style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
        >
          <div
            class="pf-v5-c-pagination__total-items"
          >
            <b>
              1
               - 
              3
            </b>
             
            of
             
            <b>
              3
            </b>
             
          </div>
          <div>
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
              id="pagination-options-menu-top-top-toggle"
              type="button"
            >
              <span
                class="pf-v5-c-menu-toggle__text"
              >
                <b>
                  1
                   - 
                  3
                </b>
                 
                of
                 
                <b>
                  3
                </b>
                 
              </span>
              <span
                class="pf-v5-c-menu-toggle__controls"
              >
                <span
                  class="pf-v5-c-menu-toggle__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <nav
            aria-label="Pagination"
            class="pf-v5-c-pagination__nav"
          >
            <div
              class="pf-v5-c-pagination__nav-control"
            >
              <button
                aria-disabled="true"
                aria-label="Go to previous page"
                class="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="previous"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe="true"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="pf-v5-c-pagination__nav-control"
            >
              <button
                aria-disabled="true"
                aria-label="Go to next page"
                class="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="next"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe="true"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </button>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </div>
  <div
    class="pf-v5-c-toolbar__content"
  >
    <div
      class="pf-v5-c-toolbar__content-section"
    >
      <div
        class="pf-v5-c-toolbar__item pf-m-chip-group pf-m-spacer-none"
      />
    </div>
  </div>
  <div
    class="pf-v5-c-toolbar__content pf-m-hidden"
    hidden=""
  >
    <div
      class="pf-v5-c-toolbar__group"
    />
  </div>
</div>
`;
